cpp/derivation.nix

```
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, cmake
+, SDL2
+, autoPatchelfHook
+, pkg-config
+, gitignoreSource
+, clang-format ? null
+, ltoSupport ? false
+, debugSupport ? false
+}:
+
+stdenv.mkDerivation rec {
+  name = "rosettaboy-cpp";
+
+  src = gitignoreSource ./.;
+
+  buildInputs = [ SDL2 ];
+  nativeBuildInputs = [ cmake pkg-config ]
+    ++ lib.optional (!stdenv.isDarwin) autoPatchelfHook;
```

Whoops, forgot to remove this one; `autoPatchelfHook` doesn't appear to be needed here either (tested on `aarch64-linux`):

```suggestion
  nativeBuildInputs = [ cmake pkg-config ];
```